### PR TITLE
Implements API to create & basic-details routes

### LINKED ISSUE #593: Implement social-links & copyright API to /edit/basic-details & create route
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend
 

Parent Issue #462

### PR DESCRIPTION
**Pull request by geekyd**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Adds missing API to basic-details & create route
- Removes route actions

Fixes #593 , #709 , #685

### CODE REVIEW

File: app/routes/events/view/edit/basic-details.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 
 export default Route.extend(EventWizardMixin, {
   titleToken() {
     return this.l10n.t('Basic Details');
   },
   model() {
-    return this.modelFor('events.view.edit');
+    let data = this.modelFor('events.view.edit');
+    data.copyright = this.getOrCreate(data.event, 'copyright', 'event-copyright');
+    data.tax = this.getOrCreate(data.event, 'tax', 'tax');
```
Comment (geekyd) [inline app/routes/events/view/edit/basic-details.js]: @niranjan94 This results in rejection of promise even if the tax exists for the event. Throws an error
```
Assertion Failed: You cannot add a record of modelClass 'undefined' to the 'event.tax' relationship (only 'tax' allowed)
```

Works perfectly for copyright & speakers call. Can you please help?

Comment (niranjan94) [inline app/routes/events/view/edit/basic-details.js]: Will check

**Comment by niranjan94** [comment]
Deployment link: https://pr-661-open-event-frontend-fossasia.surge.sh

**Comment by geekyd** [comment]
@CosmicCoder96 I forgot to mention that, the model is getting loaded again on the create route before transition. Do not know why that's happening. You have any idea?

**Comment by CosmicCoder96** [comment]
@geekyd I can't figure it out for now, but rest of it is working great. Amazing Work, how about we merge this for now, it will make creating a lot of things really easy for other PRs.

**Comment by sumedh123** [comment]
@geekyd @CosmicCoder96 I support for merge, but we will have to keep track of it.

**Comment by geekyd** [comment]
@CosmicCoder96 @sumedh123 Sure but there is one thing missing, just noticed. I will update it & we can merge this :)

**Comment by geekyd** [comment]
@CosmicCoder96 @sumedh123 @harshitagupta30 Please review.

**Comment by niranjan94** [comment]
Deployment link: https://pr-661-open-event-frontend-fossasia.surge.sh

**Pull request merged**